oauth2/oauth2-resource-server/src/test/java/org/springframework/security/oauth2/resource/web/DefaultBearerTokenResolverTests.java

```diff
@@ -0,0 +1,142 @@
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package org.springframework.security.oauth2.resource.web;
+
+import java.util.Base64;
+
+import org.junit.Before;
+import org.junit.Rule;
+import org.junit.Test;
+import org.junit.rules.ExpectedException;
+
+import org.springframework.mock.web.MockHttpServletRequest;
+import org.springframework.security.oauth2.resource.BearerTokenAuthenticationException;
+
+import static org.assertj.core.api.Assertions.assertThat;
+
+/**
+ * Tests for {@link DefaultBearerTokenResolver}.
+ *
+ * @author Vedran Pavic
+ */
+public class DefaultBearerTokenResolverTests {
+
+	private static final String TEST_TOKEN = "test-token";
+
+	@Rule
+	public ExpectedException thrown = ExpectedException.none();
+
+	private DefaultBearerTokenResolver resolver;
+
+	@Before
+	public void setUp() {
+		this.resolver = new DefaultBearerTokenResolver();
+	}
+
+	@Test
+	public void resolveWhenValidHeaderIsPresentThenTokenIsResolved() {
+		MockHttpServletRequest request = new MockHttpServletRequest();
+		request.addHeader("Authorization", "Bearer " + TEST_TOKEN);
+
+		assertThat(this.resolver.resolve(request)).isEqualTo(TEST_TOKEN);
+	}
+
+	@Test
+	public void resolveWhenNoHeaderIsPresentThenTokenIsNotResolved() {
+		MockHttpServletRequest request = new MockHttpServletRequest();
+
+		assertThat(this.resolver.resolve(request)).isNull();
+	}
+
+	@Test
+	public void resolveWhenHeaderWithWrongSchemeIsPresentThenTokenIsNotResolved() {
+		MockHttpServletRequest request = new MockHttpServletRequest();
+		request.addHeader("Authorization", "Basic " + Base64.getEncoder().encodeToString("test:test".getBytes()));
+
+		assertThat(this.resolver.resolve(request)).isNull();
+	}
+
+	@Test
+	public void resolveWhenValidHeaderIsPresentTogetherWithFormParameterThenAuthenticationExceptionIsThrown() {
+		this.thrown.expect(BearerTokenAuthenticationException.class);
+		this.thrown.expectMessage("[invalid_request]");
+
+		MockHttpServletRequest request = new MockHttpServletRequest();
+		request.addHeader("Authorization", "Bearer " + TEST_TOKEN);
+		request.setMethod("POST");
+		request.setContentType("application/x-www-form-urlencoded");
+		request.addParameter("access_token", TEST_TOKEN);
+
+		this.resolver.resolve(request);
+	}
+
+	@Test
+	public void resolveWhenValidHeaderIsPresentTogetherWithQueryParameterThenAuthenticationExceptionIsThrown() {
+		this.thrown.expect(BearerTokenAuthenticationException.class);
+		this.thrown.expectMessage("[invalid_request]");
+
+		MockHttpServletRequest request = new MockHttpServletRequest();
+		request.addHeader("Authorization", "Bearer " + TEST_TOKEN);
+		request.setMethod("GET");
+		request.addParameter("access_token", TEST_TOKEN);
+
+		this.resolver.resolve(request);
+	}
+
+	@Test
+	public void resolveWhenFormParameterIsPresentAndSupportedThenTokenIsNotResolved() {
```

Comment:
Should this be `ThenTokenIsResolved`? Same question for two tests down.

Reply:
Its possible some test have typos in names due to c&p. Will review and amend.